core/streamly-core.cabal

```diff
@@ -309,9 +309,8 @@ library
                      -- streamly-core-streams
                      , Streamly.Internal.Data.StreamK
                      -- StreamD depends on streamly-array-types
-                     , Streamly.Internal.Data.Stream.StreamD
-                     , Streamly.Internal.Data.Stream.Common
                      , Streamly.Internal.Data.Stream
+                     , Streamly.Internal.Data.Stream.Common
```

Comment:
Modules are listed in dependency order, lower level modules come before higher level ones. This should be before Data.Stream.